lib/recipe/widgets/ingredient_form.dart

```diff
@@ -47,227 +47,236 @@ class IngredientForm extends StatelessWidget {
+                        : null,
+                  ),
+                  onChanged: (value) {
+                    context.read<RecipeBloc>().add(IngredientDescriptionChanged(
+                        ingredient: state
+                            .recipes[recipeIndex].ingredients[ingredientIndex],
+                        ingredientDescription: value,
+                        recipe: state.recipes[recipeIndex]));
+                  },
+                  textInputAction: TextInputAction.next,
                 ),
-                onChanged: (value) {
-                  context.read<RecipeBloc>().add(IngredientDescriptionChanged(
-                      ingredient: state
-                          .recipes[recipeIndex].ingredients[ingredientIndex],
-                      ingredientDescription: value,
-                      recipe: state.recipes[recipeIndex]));
-                },
-                textInputAction: TextInputAction.next,
-              ),
-              DropdownFormField<String>(
-                onEmptyActionPressed: () async {},
-                decoration: InputDecoration(
-                  suffixIcon: const Icon(Icons.arrow_drop_down),
-                  icon: const Icon(Icons.food_bank_rounded),
-                  labelText: "Cooking state",
-                  helperText: 'How the ingredient is prepared',
-                  errorText: state.recipes[recipeIndex]
-                          .ingredients[ingredientIndex].cookingState.invalid
-                      ? 'Enter a cooking state'
-                      : null,
+                DropdownFormField<String>(
+                  onEmptyActionPressed: () async {},
+                  decoration: InputDecoration(
+                    suffixIcon: const Icon(Icons.arrow_drop_down),
+                    icon: const Icon(Icons.food_bank_rounded),
+                    labelText: "Cooking state",
+                    helperText: 'How the ingredient is prepared',
+                    errorText: state.recipes[recipeIndex]
+                            .ingredients[ingredientIndex].cookingState.invalid
+                        ? 'Enter a cooking state'
+                        : null,
+                  ),
+                  onSaved: (dynamic str) {}, // investigate onSaved vs onChanged
+                  onChanged: (dynamic value) {
+                    context.read<RecipeBloc>().add(
+                        IngredientCookingStateChanged(
+                            ingredient: state.recipes[recipeIndex]
+                                .ingredients[ingredientIndex],
+                            cookingState: value,
+                            recipe: state.recipes[recipeIndex]));
+                  },
+                  validator: (dynamic str) {},
+                  displayItemFn: (dynamic item) => Text(state
+                      .recipes[recipeIndex]
+                      .ingredients[ingredientIndex]
+                      .cookingState
+                      .value),
```

Comment:
@sshazril I'll have to delve into your DropdownFormFields in the future as they look a lot better than mine 😄.